### DISCUSSION TITLE
VirtualPixelMethod.Mirror

### DISCUSSION THREAD

**Discussion by AN3Orik** [dlemstra/Magick.NET#1498] [Help] [answered]
Hi!
I'm trying to add border to image and fill this border with image mirroring data via:

`image.AddBorder(20, VirtualPixelMethod.Mirror)`

```
public static MagickImage AddBorder(this MagickImage image, int borderSize, VirtualPixelMethod pixelMethod)
{
	image.VirtualPixelMethod = pixelMethod;
	image.Extent(image.Width + 2 * borderSize, image.Height + 2 * borderSize, Gravity.Center, MagickColors.Transparent);
	return image;
}
```

BUT after extent image still has transparent padding pixels instead mirroring data

![image](https://github.com/dlemstra/Magick.NET/assets/6397450/cdd44cdb-a606-4dc9-8951-c9589476fbce)

What i'm doing wrong and how can if fix this?

**Comment by dlemstra** [comment]
Using `VirtualPixelMethod.Mirror` like that is not supported. It might be best to ask for help here: https://github.com/ImageMagick/ImageMagick/discussions. Make sure you include the input image and it would also help if you include the desired result. They will tell you how to do this on the command line and then if you need help translating that to C# feel free to continue the conversation here.

**Comment by AN3Orik** [comment, marked as answer]
After reading some [docs](https://www.imagemagick.org/Usage/misc/#virtual-pixel), final code is little tricky (for my opinion):

```
public static MagickImage AddBorder(this MagickImage image, int borderSize, VirtualPixelMethod pixelMethod)
{
	image.VirtualPixelMethod = pixelMethod;
	image.FilterType = FilterType.Point;
	image.SetArtifact("distort:viewport", $"{image.Width + borderSize * 2}x{image.Height + borderSize * 2}-{borderSize}-{borderSize}");
	image.Distort(DistortMethod.ScaleRotateTranslate, 0);
	image.RemoveArtifact("distort:viewport");
	image.RePage();
	return image;
}
```
